lib/cache/senders/impressions_sender.rb

```diff
@@ -18,20 +18,30 @@ def call
           if ENV['SPLITCLIENT_ENV'] == 'test'
             post_impressions
           else
-            Thread.new do
-              @config.logger.info('Starting impressions service')
+            impressions_thread
 
-              loop do
-                post_impressions
-
-                sleep(randomize_interval(@config.impressions_refresh_rate))
+            if defined?(PhusionPassenger)
```

Comment:
@patricioe I thought it over and maybe that's not the best idea to do so, because it would mean that we'd need to provide direct access to impressions/metrics threads to the users of the SDK and make them run those threads manually in case of Passenger, seems kind of breaking of encapsulation for me. I'd say it would be better to leave it this way